internal/core/model.rs

```diff
@@ -1226,3 +1249,253 @@ fn test_vecmodel_set_vec() {
     assert_eq!(*view.reset.borrow(), 1);
     view.clear();
 }
+
+mod proxy {
+    use super::*;
+
+    pub struct MapProxy<M, F> {
+        inner_model: M,
+        map_function: F,
+    }
+
+    impl<M, F, T, U> Model for MapProxy<M, F>
+    where
+        M: 'static,
+        F: 'static,
+        F: Fn(T) -> U,
+        M: Model<Data = T>,
+    {
+        type Data = U;
+
+        fn row_count(&self) -> usize {
+            self.inner_model.row_count()
+        }
+
+        fn row_data(&self, row: usize) -> Option<Self::Data> {
+            self.inner_model.row_data(row).map(|x| (self.map_function)(x))
+        }
+
+        fn model_tracker(&self) -> &dyn ModelTracker {
+            self.inner_model.model_tracker()
+        }
+
+        fn as_any(&self) -> &dyn core::any::Any {
+            self
+        }
+    }
+
+    impl<M, F, T, U> MapProxy<M, F>
+    where
+        M: 'static,
+        F: 'static,
+        F: Fn(T) -> U,
+        M: Model<Data = T>,
+    {
+        pub fn new(model: M, map_function: F) -> Self {
+            Self { inner_model: model, map_function }
+        }
+    }
+
+    #[test]
+    fn test_map_proxy() {
+        let inner_rc = Rc::new(VecModel::from(vec![1, 2, 3]));
+        //let inner = ModelRc::from(inner_rc.clone());
+        let map = MapProxy::new(inner_rc.clone(), |x| x.to_string());
+
+        inner_rc.set_row_data(2, 42);
+        inner_rc.push(4);
+
+        assert_eq!(map.row_data(2).unwrap(), "42");
+        assert_eq!(map.row_data(3).unwrap(), "4");
+        assert_eq!(map.row_data(1).unwrap(), "2");
+    }
+
+    pub struct FilterProxy<M, F, T>
+    where
+        M: Model<Data = T> + 'static,
+        F: Fn(&T) -> bool + 'static,
+        T: 'static,
+    {
+        inner_model: M,
+        filter_function: F,
+        mapping: RefCell<Vec<usize>>,
+        notify: ModelNotify,
+    }
+
+    impl<M, F, T> FilterProxy<M, F, T>
+    where
+        M: Model<Data = T> + 'static,
+        F: Fn(&T) -> bool + 'static,
+        T: 'static,
+    {
+        fn new(inner_model: M, filter_function: F) -> Pin<Rc<RefCell<Self>>> {
```

Comment:
I think it would be best if the public API does not return a Pin.
If that means that the FilterProxy is should in fact be a wrapper to `Pin<Rc<Refcell<FilterProxyInner>>`